oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/EventHookCallback.java

```diff
@@ -0,0 +1,154 @@
+ *
+ */
+public class EventHookCallback implements AlarmCallback {
+
+    private final EventAnalyzerService analyzerService;
+
+    public EventHookCallback(ModuleManager manager) {
+        this.analyzerService = manager.find(EventAnalyzerModule.NAME).provider().getService(EventAnalyzerService.class);
+    }
+
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessage) {
+        alarmMessage.forEach(a -> {
+            for (Event event : constructCurrentEvent(a)) {
+                if (Objects.nonNull(event)) {
+                    this.analyzerService.analyze(event);
+                }
+            }
+        });
+    }
+
+    private Event[] constructCurrentEvent(AlarmMessage msg) {
+        Event[] events = new Event[2];
+        long millis = System.currentTimeMillis();
+        Event.Builder builder = Event.newBuilder()
+                .setUuid(UUID.randomUUID().toString())
+                .setName("Alarm")
+                .setStartTime(millis - msg.getPeriod())
+                .setMessage(msg.getAlarmMessage())
+                .setType(Type.Error)
+                .setEndTime(millis);
+        switch (msg.getScopeId()) {
+            case DefaultScopeDefine.SERVICE :
+                IDManager.ServiceID.ServiceIDDefinition singleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                    Source.newBuilder()
+                            .setService(singleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                break;
+            case DefaultScopeDefine.SERVICE_RELATION :
+                IDManager.ServiceID.ServiceIDDefinition doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                        Source.newBuilder()
+                            .setService(doubleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId1());
+                builder.setSource(
+                        Source.newBuilder()
+                                .setService(doubleServiceIdDef.getName())
+                                .build()
+                ).setUuid(UUID.randomUUID().toString());
+                events[1] = builder.build();
+                break;
+            case DefaultScopeDefine.SERVICE_INSTANCE :
+                IDManager.ServiceInstanceID.InstanceIDDefinition singleInstanceIdDef = IDManager.ServiceInstanceID.analysisId(msg.getId0());
+                builder.setSource(
+                        Source.newBuilder()
+                                .setServiceInstance(singleInstanceIdDef.getName())
+                                .setService(singleInstanceIdDef.getServiceId())
```

Comment:
You need to parse the service name from the ID instead of setting the ID directly